pipelines/validate-fbc-images-resolvable/validate-fbc-images-resolvable.yaml

```diff
@@ -0,0 +1,134 @@
+            type: string
+        results:
+          - name: fbc-image
+          - name: source-git-url
+          - name: source-git-revision
+        steps:
+          - name: parse
+            image: quay.io/konflux-ci/konflux-test:v1.4.26
+            script: |
+              #!/usr/bin/env bash
+              set -euo pipefail
+
+              # The SNAPSHOT parameter is not always valid JSON, so we use string manipulation
+              # to extract the required fields instead of jq.
+              echo "--- Raw SNAPSHOT parameter ---"
+              # We need to disable shellcheck for this line because Tekton will substitute the value.
+              # We actually want to print the substituted value but shellcheck doesn't know that.
+              # shellcheck disable=SC2005
+              echo "$(params.SNAPSHOT)"
+              echo "----------------------------"
+
+              # Extract values using grep and sed for robustness
+              FBC_IMAGE=$(grep -o 'containerImage:[^,}]*' <<< "$(params.SNAPSHOT)" | sed 's/containerImage://')
+              SOURCE_URL=$(grep -o 'url:[^,}]*' <<< "$(params.SNAPSHOT)" | sed 's/url://')
+              SOURCE_REV=$(grep -o 'revision:[^,}]*' <<< "$(params.SNAPSHOT)" | sed 's/revision://')
+
+              echo -n "$FBC_IMAGE" > "$(results.fbc-image.path)"
+              echo -n "$SOURCE_URL" > "$(results.source-git-url.path)"
+              echo -n "$SOURCE_REV" > "$(results.source-git-revision.path)"
+              
+              echo "Parsed FBC Image: $(cat "$(results.fbc-image.path)")"
+              echo "Parsed Git URL: $(cat "$(results.source-git-url.path)")"
+              echo "Parsed Git Revision: $(cat "$(results.source-git-revision.path)")"
+      params:
+        - name: SNAPSHOT
+          value: $(params.SNAPSHOT)
+    - name: download-idms-file
+      runAfter: [ "parse-metadata" ]
+      taskRef:
+        resolver: "git"
+        params:
+          # TODO: Before merging, change the url and revision back to the upstream repo and branch
+          - name: url
+            value: https://github.com/fontivan/konflux-ci-community-catalog.git
+          - name: revision
+            value: sskeard/validate-fbc-images-resolvable-pipeline
+          - name: pathInRepo
+            value: tasks/download-file-from-git-repo/download-file-from-git-repo.yaml
+      params:
+        - name: url
+          value: $(tasks.parse-metadata.results.source-git-url)
+        - name: revision
+          value: $(tasks.parse-metadata.results.source-git-revision)
+        - name: pathInRepo
+          value: $(params.PATH_TO_MIRROR_SET)
+        - name: REPO_TOKEN
+          value: $(params.REPO_TOKEN)
+        - name: REPO_KEY
+          value: $(params.REPO_KEY)
+    - name: run-fbc-validation
+      runAfter: [ "download-idms-file" ]
+      taskRef:
+        resolver: "git"
+        params:
+          # TODO: Before merging, change the url and revision back to the upstream repo and branch
```

Comment:
@johnbieren @Paul123111 is there any way to have this resolve to itself automatically or do I just need to update it manually before merging?

Reply:
AFAIK nothing is going to automatically update this, but I can't think of anything preventing you from updating it to this repo's url